### PR TITLE
Convert docs to JupyterBook.

### PR DESCRIPTION
**Pull request by HumairAK**
As per: https://github.com/operate-first/SRE/issues/424#issuecomment-972922870 

Check out the built JupyterBook here: https://humairak.github.io/apps/content/README.html

**What this PR does:** 
- Adds a JupyterBook to the `docs` folder, 
- Converts language to refer to these as "GitOps Docs", and that this documentation is specifically meant for GitHub Contributors interacting with the Apps repo
- JupyterBook warns against non-consecutive header increases, i.e. going from "#header" followed by "###header2" is advised against, so I've converted all headers to increase consecutively in all docs to eliminate these warnings 
- I've also added a README.md for all sub directories, JupyterBook uses the first `#header` in these markdowns as the title on the table of contents on the left panel 
- Added a contribution guide 
- Added a github action that will build this website whenever a change to the `docs/*` folder is pushed, the contents of the build are pushed to `gh-pages` branch 
- Minor updates to some headers to shorten them so that they fit neatly in the ToC on the website 

**Consequences of this PR:** 
- Nothing changes much when adding a new markdown, now instead of `apps/docs` the markdowns go in `apps/docs/content`
- To add your markdown to the JupyterBook, you just need to update [this file](https://github.com/HumairAK/apps/blob/jupyterbook/docs/_toc.yml), it's pretty self explanatory how to add new files 
- No longer need to interact with the OperateFirst website when adding docs to `apps` repo. We can just link this JupyterBook from the Operate First website.
- A lot easier to work with jupyterbook than the operate first website (quick build times as well)
- We get some nice validations from jupyterbook builds, for instance we get warnings when a markdown isn't included in a toc but is in the `docs` folder, and non-consecutive headers as mentioned earlier. 
- We get nice search bar :) 

**Next steps:** 
- Replace the apps repo docs in the operate first website with a link to this jupyterbook 
- Move hitchhiker notebooks to this jupyterbook (and archive hitchhiker repo)

### CODE REVIEW

File: docs/content/observatorium/README.md
Comment (4n4nd) [inline docs/content/observatorium/README.md:5]: looks like some kfp docs got mixed up here

File: .github/workflows/book.yml
Comment (durandom) [inline .github/workflows/book.yml:12]: i.e. the book get's rebuilt and deployed after a PR? Neat...

**Review by durandom — APPROVED** [review]
I ❤️ this. It's a great next iteration into the the right direction :)

File: .github/workflows/book.yml
Comment (HumairAK) [reply .github/workflows/book.yml:12]: yea!

**Review by 4n4nd — APPROVED** [review]
/lgtm

File: .github/workflows/book.yml
Comment (codificat) [inline .github/workflows/book.yml:38]: A suggestion: maybe publish to a [separate branch](https://github.com/peaceiris/actions-gh-pages/blob/main/README.md#%EF%B8%8F-set-another-github-pages-branch-publish_branch) instead?

```suggestion
          publish_branch: gh-pages
```

I beileve this would help keep the master branch (or any "code" branch) cleaner.

File: .github/workflows/book.yml
Comment (tumido) [inline .github/workflows/book.yml:13]: This may be worth applying here:

https://github.com/operate-first/hitchhikers-guide/pull/116/files#diff-ac885d73acfc21da974eedcc7ba75a79eca11951fb4a9afbd247a02a44c43bb4R10-R12


```
    concurrency:
      group: name
      cancel-in-progress: true
```

File: .github/workflows/book.yml
Comment (quaid) [inline .github/workflows/book.yml:14]: For my education, is this the container being used?

Is it normal for this kind of dependency to arise in a cloud native environment? Is this inherited from using an upstream project and it's not pragmatic to move everything to a standard container environment?

Or am I completely on the wrong rabbit trail here?

File: docs/CONTRIBUTING.md
Comment (quaid) [inline docs/CONTRIBUTING.md:5]: "Fork and clone" could be a link to those instructions in the Community Handbook, but we don't quite have that page. Instead we have my page on using `o1-clone` instead (https://github.com/operate-first/community-handbook/blob/main/content/how-to-use-o1-clone-tool.md).

Do we have an existing page that tells how to fork and clone as we want them done? E.g. the instructions for the website?

(not a blocking review comment)

File: docs/content/README.md
Comment (quaid) [inline docs/content/README.md:23]: I was trying to make a pull request of small changes and not only did that 404 on me with the GH API but it ended up letting my commit directly to the original branch. Sorry! These aren't major edits but should be reviewed by you before committing. Next time we'll see if I can do this correctly for suggesting changes with a PR rather than here in the comments.

File: docs/content/about/intro.md
Comment (quaid) [inline docs/content/about/intro.md:28]: List should be https://lists.operate-first.cloud/archives/list/community@lists.operate-first.cloud/

File: docs/CONTRIBUTING.md
Comment (HumairAK) [reply docs/CONTRIBUTING.md:5]: > Do we have an existing page that tells how to fork and clone as we want them done? E.g. the instructions for the website? 

We don't, maybe we should.I think it's better to describe verbatim what we want them to do in this doc, then link to it from all other docs as part of a "Pre-requisites" section.

I'd suggest making an issue for this however as it's out of scope for this PR.

**Review by quaid — COMMENTED** [review]
I have a few questions in the comments that are about my curiosity/education and are not blockers to merging.

There are a few quick comments to address before merging, e.g. correcting the link to the mailing list.

I also was trying to do a PR of further edits to the existing PR but it only let me commit directly to the branch instead. Ideally, an editor should be able to queue up a bunch of edit suggestions for a writer as a PR rather than committing them to the branch directly. This way the writer can be the arbiter of decisions made about their own work.

**Comment by HumairAK** [comment]
@quaid the pr allows maintainers to push changes, I've disabled this feature. I'll remove your current commits to keep this pr clean. And then you can offer your suggestions as either commit suggestions or comments and I can inact them. Or save them for a follow up PR. Though I recommend saving content updates to a separate PR, I'd like to keep this PR focused on conversion to JupyterBook mostly.

File: docs/content/about/intro.md
Comment (HumairAK) [reply docs/content/about/intro.md]: updated

File: docs/content/README.md
Comment (HumairAK) [reply docs/content/README.md:22]: Updated

File: .github/workflows/book.yml
Comment (HumairAK) [reply .github/workflows/book.yml:14]: This specifies the virtual environment for the github action. You can read more about these here: https://github.com/actions/virtual-environments

File: .github/workflows/book.yml
Comment (HumairAK) [reply .github/workflows/book.yml:13]: neat, done!

File: docs/CONTRIBUTING.md
Comment (larsks) [inline docs/CONTRIBUTING.md:12]: This should probably just read "Update `docs/_toc.yml`" instead of "Update the `docs/_toc.yml`" (or "Update the `docs/_toc.yml` file"). Also, most of our YAML files in this repository are `.yaml`. Any particular reason why `_toc.yml` is different? Not a problem; I just like consistency :).

I think "markdown" should be an adjective, not a noun, and I think you don't need to use `pre-formatted` text here other than for the filename:

> Update `docs/_toc.yml` by adding the location of your new markdown document to the appropriate chapter/section.

File: docs/CONTRIBUTING.md
Comment (larsks) [inline docs/CONTRIBUTING.md:16]: I would make "`apps` repo" a link to the apps repo (you've already defined the `apps-repo` link):

```
...PR to the [`apps` repo][apps-repo]
```

File: docs/CONTRIBUTING.md
Comment (larsks) [inline docs/CONTRIBUTING.md:20]: Missing the indefinite article:

> ...you can create **a** virtual environment...

File: docs/CONTRIBUTING.md
Comment (larsks) [inline docs/CONTRIBUTING.md:24]: Definite article and maybe a minor change in phrasing:

> ...you can enter **the virtual environment**

File: docs/CONTRIBUTING.md
Comment (larsks) [inline docs/CONTRIBUTING.md:28]: Definite article and capitalization:

> Now build **the Jupyter** book

File: docs/CONTRIBUTING.md
Comment (larsks) [inline docs/CONTRIBUTING.md:33]: Missing the definite article:

> ...can be found in **the** `_build` folder.

**Review by larsks — CHANGES_REQUESTED** [review]

File: docs/CONTRIBUTING.md
Comment (HumairAK) [reply docs/CONTRIBUTING.md:12]: > Also, most of our YAML files in this repository are .yaml. Any particular reason why _toc.yml is different? Not a problem; I just like consistency :). 

I was getting ready to say "no particular reason", but decide to run the jupyterbook with these files as *.yaml, and it crashed, lol. 

```


Couldn't find a Table of Contents file. To auto-generate one, run:
        jupyter-book toc from-project .
```

I guess you have to specify it manually if you use anything other than `_toc.yml`...I'm inclined to just leave it as is (unless you feel strongly here :) )... otherwise I'd have to also change it in the action.

Comment (larsks) [reply docs/CONTRIBUTING.md:12]: I don't feel strongly :).

**Review by larsks — APPROVED** [review]

**Comment by quaid** [comment]
> Though I recommend saving content updates to a separate PR, I'd like to keep this PR focused on conversion to JupyterBook mostly.

Thanks, I concur, I prefer to keep content edits away from other edits. In the past I've had success with PRs within PRs like I tried here, but it's a bit confusing and not clean, process wise.

Let's establish from now on that for content edits we can follow an editorial flow that separates out these edits from other PR reviews. That is implied in the process, but not made clear what to do when there is an edit to Community Handbook content within the body of a PR full of other changes.

Your proposal is to do content edits separately and after the PR is merged. I'm fine with that myself because it's relatively easier than having someone like you separate out your docs changes to a different PR.

https://github.com/operate-first/community-handbook#how-to-contribute

I created this issue for tracking any changes to the Handbook workflow process we've identified here, then making those changes.  https://github.com/operate-first/community-handbook/issues/25

**Pull request merged**
